### PR TITLE
fix: make all tests green with new view engine and JIT

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/@angular/common/test/directives/ng_class_spec.ts [hunk 1]
```diff
@@ -13,9 +13,12 @@ export function main() {
   describe('binding to CSS class list', () => {
     let fixture: ComponentFixture<any>;
 
+    function normalizeClassNames(classes: string) { return classes.split(' ').sort().join(' '); }
+
     function detectChangesAndExpectClassName(classes: string): void {
       fixture.detectChanges();
-      expect(fixture.debugElement.children[0].nativeElement.className.trim()).toEqual(classes);
+      let nonNormalizedClassName = fixture.debugElement.children[0].nativeElement.className;
```
Comment (tbosch) [inline modules/@angular/common/test/directives/ng_class_spec.ts]: add `.trim()` again here.

File: modules/@angular/common/test/directives/ng_class_spec.ts [hunk 2]
```diff
@@ -13,9 +13,12 @@ export function main() {
   describe('binding to CSS class list', () => {
     let fixture: ComponentFixture<any>;
 
+    function normalizeClassNames(classes: string) { return classes.split(' ').sort().join(' '); }
```
Comment (tbosch) [inline modules/@angular/common/test/directives/ng_class_spec.ts]: Add trim here

File: modules/@angular/compiler/src/view_compiler_next/view_compiler.ts [hunk 3]
```diff
@@ -493,7 +503,7 @@ class ViewBuilder implements TemplateAstVisitor, LocalResolver, BuiltinConverter
         outputDefs.push(new o.LiteralMapEntry(propName, o.literal(eventName), false));
       }
     });
-    if (directiveAst.inputs.length) {
+    if (directiveAst.inputs.length || directiveAst.directive.type.lifecycleHooks.length > 0) {
```
Comment (tbosch) [inline modules/@angular/compiler/src/view_compiler_next/view_compiler.ts]: check for `NgDoCheck`

File: modules/@angular/core/src/view/refs.ts [hunk 4]
```diff
@@ -207,3 +251,137 @@ class Injector_ implements Injector {
...
+      return asProviderData(view, def.index).instance;
+  }
+  return undefined;
+}
+
+export function createRendererV1(view: ViewData): RendererV1 {
+  return new RendererV2V1Adatper(view.renderer);
+}
+
+class RendererV2V1Adatper implements RendererV1 {
```
Comment (tbosch) [inline modules/@angular/core/src/view/refs.ts]: `RendererAdapter`

**Review by vicb — APPROVED** [review]

File: modules/@angular/common/test/directives/ng_class_spec.ts (hunk 2 above)
Comment (tbosch) [inline modules/@angular/common/test/directives/ng_class_spec.ts]: Done

File: modules/@angular/compiler/src/view_compiler_next/view_compiler.ts (hunk 3 above)
Comment (tbosch) [inline modules/@angular/compiler/src/view_compiler_next/view_compiler.ts]: Done

File: modules/@angular/core/src/view/refs.ts (hunk 4 above)
Comment (tbosch) [inline modules/@angular/core/src/view/refs.ts]: done

**Review by IgorMinar — APPROVED** [review]

**Pull request merged**